transaction/core/src/tx.rs

```diff
@@ -316,21 +318,13 @@ impl TxOutMembershipProof {
     /// # Arguments
     /// * `index` - The index of the TxOut.
     /// * `highest_index` - The index of the last TxOut in the ledger, indicating the size of the tree that the proof refers to.
-    /// * `range_to_hash` - Mapping from a range of TxOut indices to the hash "over" those outputs.
-    pub fn new(index: u64, highest_index: u64, range_to_hash: HashMap<Range, [u8; 32]>) -> Self {
-        let mut hashes: Vec<TxOutMembershipElement> = range_to_hash
-            .into_iter()
-            .map(|(r, h)| TxOutMembershipElement {
-                range: r,
-                hash: TxOutMembershipHash(h),
-            })
-            .collect();
-        hashes.sort();
-
+    /// * `elements` - The tx out membership elements, containing ranges referring to subtrees in the tree, and hashes.
+    ///                These must be provided in the order in which they should be combined to validate the proof.
+    pub fn new(index: u64, highest_index: u64, elements: Vec<TxOutMembershipElement>) -> Self {
```

Comment:
the most significant API change is that new takes the sequence of elements in the right order, and doesn't throw them into a hashmap or force the caller to do that. we want to keep them in the right order, which is the same order that they are generated in

Reply:
This is great.